src/BlazorWebView/src/Maui/Windows/WinUIWebViewManager.cs

```diff
@@ -81,35 +101,54 @@ protected override async Task HandleWebResourceRequest(CoreWebView2WebResourceRe
 				var headerString = GetHeaderString(headers);
 				eventArgs.Response = _coreWebView2Environment!.CreateWebResourceResponse(ms, statusCode, statusMessage, headerString);
 			}
-			else
+			else if (new Uri(requestUri) is Uri uri && new Uri(AppOrigin).IsBaseOf(uri))
 			{
-				// Next, try to go through WinUI Storage to find a static web asset
-				var uri = new Uri(requestUri);
-				if (new Uri(AppOrigin).IsBaseOf(uri))
-				{
-					var relativePath = new Uri(AppOrigin).MakeRelativeUri(uri).ToString();
-
-					// If the path does not end in a file extension (or is empty), it's most likely referring to a page,
-					// in which case we should allow falling back on the host page.
-					if (allowFallbackOnHostPage && !Path.HasExtension(relativePath))
-					{
-						relativePath = _hostPageRelativePath;
-					}
-					relativePath = Path.Combine(_contentRootDir, relativePath.Replace('/', '\\'));
+				var relativePath = new Uri(AppOrigin).MakeRelativeUri(uri).ToString();
 
+				// If the path does not end in a file extension (or is empty), it's most likely referring to a page,
+				// in which case we should allow falling back on the host page.
+				if (allowFallbackOnHostPage && !Path.HasExtension(relativePath))
+				{
+					relativePath = _hostPageRelativePath;
+				}
+				relativePath = Path.Combine(_contentRootDir, relativePath.Replace('/', '\\'));
+				statusCode = 200;
+				statusMessage = "OK";
+				var contentType = StaticContentProvider.GetResponseContentTypeOrDefault(relativePath);
+				headers = StaticContentProvider.GetResponseHeaders(contentType);
+				var headerString = GetHeaderString(headers);
+				IRandomAccessStream? stream = null;
+				if (_isPackagedApp)
+				{
 					var winUIItem = await Package.Current.InstalledLocation.TryGetItemAsync(relativePath);
 					if (winUIItem != null)
 					{
-						statusCode = 200;
-						statusMessage = "OK";
-						var contentType = StaticContentProvider.GetResponseContentTypeOrDefault(relativePath);
-						headers = StaticContentProvider.GetResponseHeaders(contentType);
-						var headerString = GetHeaderString(headers);
-						var stream = await Package.Current.InstalledLocation.OpenStreamForReadAsync(relativePath);
-
-						eventArgs.Response = _coreWebView2Environment!.CreateWebResourceResponse(stream.AsRandomAccessStream(), statusCode, statusMessage, headerString);
+						var contentStream = await Package.Current.InstalledLocation.OpenStreamForReadAsync(relativePath);
+						stream = contentStream.AsRandomAccessStream();
 					}
 				}
+				else
+				{
+					var path = Path.Combine(Environment.CurrentDirectory, relativePath);
```

Comment:
How do we know that `Environment.CurrentDirectory` is correct? It's mutable, so that's always a concern, but how do we know that it's correct even at startup? Is it always set to the app root in WinUI unpackaged apps? (That seems very unlikely to me.)

Reply:
`Environment.CurrentDirectory` is better than `Directory.GetCurrentDirectory()`. Another alternative could be `Assembly.GetEntrypointAssembly().Location` but I do have similar concerns about it being right. I think `Appcontext.BaseDirectory` might do the trick.

In general I'm not super concerned about using one or the other `Environment.CurrentDirectory` is technically preferred to `Directory.GetCurrentDirectory()`. AppContext.BaseDirectory and `Assembly.GetEntrypointAssembly().Location` both assume that the wwwroot folder is side by side with the loaded assembly

Reply:
I have news to you about the two Current Directory APIs: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/IO/Directory.cs#L212 😁 

And I am quite sure that using it is wrong, because there's generally no guarantee that it is what we want even when the app starts, and because it's mutable, there's no guarantee what it ever is at any time.

`AppContext.BaseDirectory` sounds more interesting, but it is also not _semantically_ what we want, though I agree it is super likely to be correct. And, it's immutable, and set "outside" of the logic of the app (that is, the app doesn't decide for itself once it's running what the value should be). Same story with `Assembly.GetEntrypointAssembly().Location`.

How will the rest of MAUI figure out from where to load assets? Presumably the same exactly logic will apply, no? I think ``AppContext.BaseDirectory` likely the safest bet but we also need to be consistent with all of MAUI.

Reply:
```suggestion
					var path = Path.Combine(AppContext.BaseDirectory, relativePath);
```